server/market/market.go

```diff
@@ -1992,8 +1992,9 @@ func (m *Market) handlePreimageResp(msg *msgjson.Message, reqData *piData) {
 	if err != nil {
 		log.Errorf("StorePreimage: %v", err)
 		// Fatal backend error. New swaps will not begin, but pass the preimage
-		// along so it does not appear as a miss to collectPreimages.
-		m.respondError(msg.ID, reqData.ord.User(), msgjson.UnknownMarketError, "internal server error")
+		// along so that it does not appear as a miss to collectPreimages.
+		m.respondError(msg.ID, reqData.ord.User(), msgjson.UnknownMarketError,
+			fmt.Sprintf("internal server error: %v", err))
```

Comment:
I'm assuming here there is "nothing to hide" from the client; in other words, this error shouldn't contain sensitive information, thoughts ?

Reply:
Also, seems like `msgjson.UnknownMarketError` code here (and in some other similar places) isn't the best fit ? Thought I'd call this out, and maybe fix.

Reply:
So, would it be ok to replace `msgjson.UnknownMarketError` (and in similar cases where appropriate) with `msgjson. RPCInternalError` ? and revert back `fmt.Sprintf("internal server error: %v", err)` -> `"internal server error"`,

> The error is likely to be from postgres here, and sending it blindly to the client might leak info about server config

since looks like this is applicable only to "internal" cases calling `respondError` I've modified in this PR

Reply:
That sounds good.  The other unmarshal errors and similar can still go back to the client though I think.

Reply:
Done.